src/main/java/com/sekwah/advancedportals/bukkit/listeners/PluginMessageReceiver.java

```diff
@@ -37,14 +37,12 @@ public void onPluginMessageReceived(String channel, Player player, byte[] messag
             String targetDestination = in.readUTF();
             String bungeeUUID = in.readUTF();
 
-            Player targetPlayer = this.plugin.getServer().getPlayer(UUID.fromString(bungeeUUID));
```

Comment:
This is required as the player on this method is the player's connection being used to send the data and isn't guaranteed to be the correct user. This is due to the wording of the implementation as well as how bungee generally works. 

However, I am unsure how this works on velocity.

If this is not working for you then the UUID's may be different across the networks and I would recommend UUID forwarding from the proxy.

Reply:
Actually looking at this more this has a good chance to teleport the wrong player on active servers.